panic: unaligned 64-bit atomic operation

### Subject of the issue

algod panics. Seems like its able to commit a few blocks after each restart, then panics with: 

```
panic: unaligned 64-bit atomic operation
goroutine 565 [running]:
runtime/internal/atomic.panicUnaligned()
        runtime/internal/atomic/unaligned.go:8 +0x24
runtime/internal/atomic.Xadd64(0xee3d10c, 0x1)
        runtime/internal/atomic/atomic_arm.s:256 +0x14
github.com/algorand/go-algorand/network.(*wsPeer).handleFilterMessage(0xee3d080, {{0xe080e0, 0xee3d080}, {0xf52def4, 0x2}, {0x103427e0, 0x20, 0x20}, {0x0, 0x0}, ...})
        github.com/algorand/go-algorand/network/wsPeer.go:589 +0xf0
github.com/algorand/go-algorand/network.(*wsPeer).readLoop(0xee3d080)
        github.com/algorand/go-algorand/network/wsPeer.go:495 +0x57c
created by github.com/algorand/go-algorand/network.(*wsPeer).init
        github.com/algorand/go-algorand/network/wsPeer.go:377 +0x3b8
```

Seems like the last logged line in node.log consistently is:
```
{"event":"Disconnected","file":"wsNetwork.go","function":"github.com/algorand/go-algorand/network.(*WebsocketNetwork).removePeer","level":"info","line":2186,"local":"","msg":"Peer r-hg.algorand-mainnet.network:4160 disconnected: ReadError","name":"","remote":"r-hg.algorand-mainnet.network:4160","time":"2022-10-09T16:56:42.369986Z"}
```

### Your environment

```
§ algod -v
12885557248
3.10.0.stable [rel/stable] (commit #9869cd96)
go-algorand is licensed with AGPLv3.0
source code available at https://github.com/algorand/go-algorand
```

Raspberry Pi 4 8GB
```
$ uname -a
Linux alarmpi 5.15.61-4-rpi-ARCH #1 SMP Thu Sep 1 13:35:37 MDT 2022 armv7l GNU/Linux
```

### Steps to reproduce

1. Start node
2. Wait a few seconds

### Expected behaviour

No panic

### Actual behaviour

panic

--- a/logging/telemetryspec/event.go
+++ b/logging/telemetryspec/event.go
@@ -301,7 +301,7 @@ type PeerConnectionDetails struct {
 	// MessageDelay is the avarage relative message delay. Not being used for incoming connection.
 	MessageDelay int64 `json:",omitempty"`
 	// DuplicateFilterCount is the number of times this peer has sent us a message hash to filter that it had already sent before.
-	DuplicateFilterCount int64
+	DuplicateFilterCount uint64
 }
 
 // CatchpointGenerationEvent event

--- a/network/wsPeer.go
+++ b/network/wsPeer.go
@@ -177,6 +177,11 @@ type wsPeer struct {
 	// peer, or zero if no message is being written.
 	intermittentOutgoingMessageEnqueueTime int64
 
+	// duplicateFilterCount counts how many times the remote peer has sent us a message hash
+	// to filter that it had already sent before.
+	// this needs to be 64-bit aligned for use with atomic.AddInt64 on 32-bit platforms.
+	duplicateFilterCount int64
+
 	// Nonce used to uniquely identify requests
 	requestNonce uint64
 
@@ -203,9 +208,6 @@ type wsPeer struct {
 
 	incomingMsgFilter *messageFilter
 	outgoingMsgFilter *messageFilter
-	// duplicateFilterCount counts how many times the remote peer has sent us a message hash
-	// to filter that it had already sent before.
-	duplicateFilterCount int64
 
 	processed chan struct{}
 

--- a/network/wsPeer.go
+++ b/network/wsPeer.go
@@ -182,8 +182,8 @@ type wsPeer struct {
 
 	// duplicateFilterCount counts how many times the remote peer has sent us a message hash
 	// to filter that it had already sent before.
-	// this needs to be 64-bit aligned for use with atomic.AddInt64 on 32-bit platforms.
-	duplicateFilterCount int64
+	// this needs to be 64-bit aligned for use with atomic.AddUint64 on 32-bit platforms.
+	duplicateFilterCount uint64
 
 	wsPeerCore
 
@@ -616,7 +616,7 @@ func (wp *wsPeer) handleFilterMessage(msg IncomingMessage) {
 		// large message concurrently from several peers, and then sent the filter message to us after
 		// each large message finished transferring.
 		duplicateNetworkFilterReceivedTotal.Inc(nil)
-		atomic.AddInt64(&wp.duplicateFilterCount, 1)
+		atomic.AddUint64(&wp.duplicateFilterCount, 1)
 	}
 }
 

--- a/network/wsPeer.go
+++ b/network/wsPeer.go
@@ -177,13 +177,13 @@ type wsPeer struct {
 	// peer, or zero if no message is being written.
 	intermittentOutgoingMessageEnqueueTime int64
 
+	// Nonce used to uniquely identify requests
+	requestNonce uint64
+
 	// duplicateFilterCount counts how many times the remote peer has sent us a message hash
 	// to filter that it had already sent before.
 	// this needs to be 64-bit aligned for use with atomic.AddInt64 on 32-bit platforms.
 	duplicateFilterCount int64
-
-	// Nonce used to uniquely identify requests
-	requestNonce uint64
 
 	wsPeerCore
 

--- a/network/wsPeer_test.go
+++ b/network/wsPeer_test.go
@@ -103,6 +103,7 @@ func TestAtomicVariablesAlignment(t *testing.T) {
 	require.True(t, (unsafe.Offsetof(p.requestNonce)%8) == 0)
 	require.True(t, (unsafe.Offsetof(p.lastPacketTime)%8) == 0)
 	require.True(t, (unsafe.Offsetof(p.intermittentOutgoingMessageEnqueueTime)%8) == 0)
+	require.True(t, (unsafe.Offsetof(p.duplicateFilterCount)%8) == 0)
 }
 
 func TestTagCounterFiltering(t *testing.T) {

